android/build.gradle

```diff
@@ -51,7 +53,8 @@ dependencies {
     implementation "com.android.installreferrer:installreferrer:${safeExtGet('installReferrerVersion', '1.1.2')}"
     def firebaseIidVersion = safeExtGet('firebaseIidVersion', null)
     if(firebaseIidVersion){
-        implementation "com.google.firebase:firebase-iid:$firebaseIidVersion"
+        implementation platform("com.google.firebase:firebase-bom:${safeExtGet("firebaseBomVersion", "25.12.2")}")
+        implementation "com.google.firebase:firebase-iid:${safeExtGet("firebaseIidVersion", "20.2.3")}"
     }else{
         def iidVersion = safeExtGet('googlePlayServicesIidVersion', safeExtGet('googlePlayServicesVersion', '17.0.0'))
         implementation "com.google.android.gms:play-services-iid:$iidVersion"
```

Comment:
This part won't work well. The IID library was dropped from the BoM for a while, so depending on what BoM you are using for other reasons (e.g., react-native-firebase) you may need to specify a BoM that does *not* have an IID version, at which point we really need to use the firebaseIidVersion the user specifies.

Here you are saying "do you have a firebaseIidVersion?" then doing BoM work when you have it

The point of the BoM is that *if* you have it, then you don't put a version for the IID dependency at all, so this does not really help anything, it just gives us new external gradle config surface area (the firebaseBoMVersion) that we aren't using, and can't use very well

https://github.com/firebase/firebase-android-sdk/issues/1077#issuecomment-657011199

Reply:
in my project with many libraries the firebase-bom does not works and makes dependencies conflict. that's why i have chosen specifying firebaseIidVersion. Anyway, i did not try to specify firebaseIidVersion with firebaseBomVersion together

Reply:
for example i have react-native-push-notification, react-native-firebase (crashlytics and messaging) and react-native-device-info and all this libs depends on firebase IID